src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp

```diff
@@ -150,13 +182,37 @@ namespace frame_calibration_node
                 geodesy_tf_msg.pose.pose.position.x = geodesy_tf_point.point.x;
                 geodesy_tf_msg.pose.pose.position.y = geodesy_tf_point.point.y;
                 geodesy_tf_msg.header.frame_id = geodesy_tf_point.header.frame_id;
+                // geodesy_tf_msg.child_frame_id = temp_geodesy_tf_point.header.frame_id;
                 geodesy_tf_msg.header.stamp = ros::Time::now();
 
-                geodesy_tf_pub.publish(geodesy_tf_msg);
+                // counter to remove edge case of first data point
+                gps_msg_counter += 1;
+                
+                if(gps_msg_counter == 1)
+                {
+                    geodesy_tf_pub.publish(geodesy_tf_msg);    
+                }
+                else
+                {
+                    // calculate the euclidian distance between point (n) and point (n-1)
+                    float distance_comparison;
+                    float gps_threshold = 0.5; // [m]

```

Comment:
Consider using a rosparam for this, so it's easily configurable from a launch file

Reply:
Ok